src/components/LoadingShape/index.js

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import RenderIf from '../RenderIf';
+import {
+    StyledShapeContainer,
+    StyledLoadingShape,
+    StyledImageIcon,
+    StyledAvatarIcon,
+} from './styled';
+
+/**
+ * LoadingShape can be used to display a placeholder where content
+ * is being loaded asynchronously.
+ */
+const LoadingShape = props => {
+    const { className, style, shape, variant } = props;
+    const shapeRef = useRef();
+    const isImage = variant === 'image';
+    const isAvatar = variant === 'avatar';
+
+    useEffect(() => {
+        const el = shapeRef.current;
+        if (shape === 'square' || shape === 'circle') {
+            el.style.width = `${el.offsetHeight}px`;
+        } else {
+            el.style.width = '100%';
+        }
+    });
```

Comment:
can we use an array of dependencies here? with the shape prop, thus it will run only when the shape prop changes